packages/react-composites/tests/browser/chat/fake-adapter/fake-back-end/ThreadEventEmitter.ts

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { CommunicationIdentifier } from '@azure/communication-common';
+import {
+  ChatMessageDeletedEvent,
+  ChatMessageEditedEvent,
+  ChatMessageReceivedEvent,
+  ChatThreadCreatedEvent,
+  ChatThreadDeletedEvent,
+  ChatThreadPropertiesUpdatedEvent,
+  ParticipantsAddedEvent,
+  ParticipantsRemovedEvent,
+  ReadReceiptReceivedEvent,
+  TypingIndicatorReceivedEvent
+} from '@azure/communication-signaling';
+import { toFlatCommunicationIdentifier } from '@internal/acs-ui-common';
+import { EventEmitter } from 'events';
+import { NetworkEventModel, Thread } from './types';
+
+export class ThreadEventEmitter {
```

Comment:
can we strongly type an interface that has all + future events?

Reply:
`export class ThreadEventEmitter implements ChatEventEmitter`
so if a new chatFILESOMETHINGReceived event, this build needs to break